### DISCUSSION TITLE
Q&A Session des Theorie und Grundlagenteils der VDI Dojos

### DISCUSSION THREAD

**Discussion by stan1025** [VDI-CodING/codeING-main#59] [General]
Liebe Coderinnen und Coder,

hier ist nochmal der Link für die in den Videos angekündigte Q&A Session heute Abend.

https://teams.microsoft.com/l/meetup-join/19%3ameeting_MDRkZmE2YTQtNjdhNy00ZGQ5LWJjMjAtNzU1NTlkZThjYmUy%40thread.v2/0?context=%7b%22Tid%22%3a%2238ae3bcd-9579-4fd4-adda-b42e1495d55a%22%2c%22Oid%22%3a%22959e1c6d-560c-4f2d-8d1b-19b1234c4fc9%22%7d

Wann: Heute 14.05.2021 von 17.00 bis 18.30 Uhr
Was: Wir beantworten eure Frage aus den Grundlagen Sections und möchten und mit euch austauschen, welche Art von Aufgaben wir nun weiter anbieten sollen?

Wir freuen uns auf euch!

**Comment by w-tusche** [comment]
Hallo, 

vielen Dank für das Anbieten eines Q&A´s. Leider kann ich gleich nicht, deswegen stelle ich meine Frage einfach hier.

In S1D4, HierarchyOrgEinheit in der Methode AlleEinheiten() gibt es den for-loop:

``` javascript
for (const val in this.Einheiten) {
    const temp = this.Einheiten[val]
    if (temp instanceof HierarchyOrgEinheit) {
      list = list.concat((temp as HierarchyOrgEinheit).AlleEinheiten())
    } else if (temp instanceof OrgEinheit) {
      list.push(temp)
    }
  }
```

Ich habe einige Schwierigekeiten hier genau zu verstehen, wie das Eintragen in die Liste bei den Unterkategorien funktioniert. Es war mir etwas viel mit dem `instanceof`, und `as`  und der Rekursion alles in ein paar Zeilen. Vielleicht könnt ihr das noch einmal etwas aufdröseln, was da genau in welcher Zeile passiert.

Vielen Danke im Vorraus!
